internal/build/image_builder.go

```diff
@@ -341,6 +341,9 @@ func readDockerOutput(ctx context.Context, reader io.Reader, writer io.Writer) (
 	span, ctx := opentracing.StartSpanFromContext(ctx, "daemon-readDockerOutput")
 	defer span.Finish()
 
+	progressLastPrinted := make(map[string]time.Time)
```

Comment:
Nit: It would be helpful for code comprehensibility if the key here was an opaque type that described what this string is. I was thinking like:

```golang
type buildkitMessageID = string
progressLastPrinted := make(map[buildkitMessageID]time.Time)
```

Reply:
done! tho it's worth pointing out that this isn't from buildkit, it's from docker push